src/presentation.ts

```diff
@@ -12,18 +13,30 @@ import {
   type StoredCredential,
 } from './credential.ts';
 import { assert } from './util.ts';
+import { generateContext, computeContext } from './program-spec.ts';
+import { NestedProvable } from './nested.ts';
 import { serializePresentationRequest } from './serialize-spec.ts';
 import { deserializePresentationRequest } from './deserialize-spec.ts';
 
 export { PresentationRequest, Presentation };
 
-type PresentationRequestType = 'no-context';
+type InputContext = {
+  type: 'zk-app' | 'https';
+  presentationCircuitVKHash: Field;
+  action: Field | string;
+  serverNonce: Field;
+};
+
+type WalletContext = {
+  verifierIdentity: string | PublicKey;
+  clientNonce: Field;
+};
+
+type PresentationRequestType = 'no-context' | 'with-context';
```

Comment:
can you make separate pres.req. types for the two different types of context? and have two different create functions for each of them? and keep the two generic type arguments to `PresentationRequest`

I see that you made it work for both at the same time, but I prefer separating them because:
* different contexts imply different security assumptions, and the wallet should be able to choose which presentation type (= context type) it even supports
* the two context types we have right now are just an early iteration, we might change them or add more, and then it's easier if we have two separate entry points
* leaving the generic type arguments in there allows the request to be typed more precisely. e.g. right now you have `action: string | Field` in the type when really it should be just `string` for "https" type requests, and `Field` for "zkapp" type requests

Reply:
i.e. it should be
```ts
type PresentationRequestType = 'no-context' | 'zk-app' | 'https';

PresentationRequest.zkapp(...): PresentationRequest;
PresentationRequest.https(...): PresentationRequest;
```